### PR TITLE
Update template Data Mapper to bridge differences between CoE templat…

### PR DESCRIPTION
**Pull request by tchow8**
…e and CCD Config

NOTE: Do not merge yet. Pending discussion.

# Description

There are a few mismatches between the configuration and the CoE template. There is a process to uploading / changing either of the two.
If the decision is made to bring both in line, this PR can be reduced.
Otherwise it is required to ensure the Docmosis Render service receives all required data mappings in the correct format.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/TemplateDataMapper.java [hunk 1]
```diff
@@ -71,8 +72,16 @@
...
-        // Setup hardcoded service centre court contact details
-        data.put(COURT_CONTACT_KEY, SERVICE_CENTRE_COURT_CONTACT_DETAILS);
+        // Setup hardcoded service centre court contact details if missing
+        if (Objects.isNull(data.get(COURT_CONTACT_KEY))) {
+            data.put(COURT_CONTACT_KEY, SERVICE_CENTRE_COURT_CONTACT_DETAILS);
+        }
+
+        // Set a default value for Who Pays Costs if not set but costs claim is granted
+        if (String.valueOf(data.get(COSTS_CLAIM_GRANTED_JSON_KEY)).equalsIgnoreCase(YES_VALUE)
+                && Objects.isNull(data.get(WHO_PAYS_COSTS_JSON_KEY))) {
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/TemplateDataMapper.java]: I thought WHO_PAYS_COSTS_JSON_KEY  is mandatory when COSTS_CLAIM_GRANTED is Yes

Comment (tchow8) [reply src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/TemplateDataMapper.java]: Verifying

Comment (tchow8) [reply src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/mapper/TemplateDataMapper.java]: Removing, Victor says he will default it to respondent.

File: src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/service/impl/DocumentManagementServiceImpl.java [hunk 2]
```diff
@@ -75,7 +76,12 @@ public GeneratedDocumentInfo storeDocument(byte[] document, String authorization
         log.debug("Generate document requested with templateName [{}], placeholders of size[{}]",
                 templateName, placeholders.size());
 
-        Map<String, Object> formattedPlaceholders = HtmlFieldFormatter.format(placeholders);
+        Map<String, Object> formattedPlaceholders = placeholders;
+
+        // Reform PDF Generator requires formatting for certain characters
+        if (PDF_GENERATOR_TYPE.equals(pdfGenerationFactory.getGeneratorType(templateName))) {
```
Comment (alexwakeman) [inline src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/service/impl/DocumentManagementServiceImpl.java]: `PDF_GENERATOR_TYPE` is a bit ambiguous I think - could be more specific like `REFORM_GENERATOR_SERVICE`

Comment (tchow8) [reply src/main/java/uk/gov/hmcts/reform/divorce/documentgenerator/service/impl/DocumentManagementServiceImpl.java]: I don't want to lose my reviews haha so I'll make a note for this change on my next PR
(Decree Nisi document generation)

**Pull request merged**